core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -494,9 +491,13 @@ public void evict(WrappedBytes key) {
       lock.lock();
       try {
          checkDeallocation();
-         // TODO: this could be more efficient
-         passivator.passivate(get(key));
-         remove(key);
+         long address = memoryLookup.getMemoryAddress(key);
+         if (address != 0) {
+            // TODO: this could be more efficient
+            InternalCacheEntry<WrappedBytes, WrappedBytes> ice = performGet(address, key);
+            passivator.passivate(ice);
```

Comment:
We should do a null check for the ice and only perform these 2 lines if it exists. This way we prevent a possible loop around the memory address linked list if it is present.